utils-intervalrdd/src/main/scala/org/bdgenomics/utils/interval/array/IntervalArray.scala

```diff
@@ -144,6 +145,36 @@ trait IntervalArray[K <: Interval[K], T] extends Serializable {
     }
   }
 
+  /**
+   * Finds an index in array that overlaps with the query rr, or if none exists
```

Comment:
+1, I don't think we need to do the clean up in this PR, but I agree with the cleanup @heuermh is proposing, and if we don't do it here, we should at least open a ticket so it doesn't get dropped.